packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryUtils.java

```diff
@@ -199,7 +204,13 @@ static void putAssetsInfo(ContentResolver contentResolver, Cursor cursor, ArrayL
       if (fullInfo) {
         if (exifInterface != null) {
           getExifFullInfo(exifInterface, asset);
-          getExifLocation(exifInterface, asset);
+
+          if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            Uri photoUri = Uri.withAppendedPath(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, cursor.getString(idIndex));
+            getExifLocationForUri(contentResolver, photoUri, asset);
+          } else {
```

Comment:
Is this block intentionally inside `if (exifInterface != null)`? As `getExifLocationForUri` does not depend on that original `exifInterface`, maybe we can extract it outside?

Reply:
it does require this check as it also requires the `getExifFullInfo` - I could extract but then would have to duplicate the logic around adding extra info - I think its a matter of taste so either way let me know what you'd prefer!

Reply:
Hmm, if _most likely_ global `null`ified "global" `ExifInterface` implies null exifInterface created by `getExifLocationForUri` (so most likely there are cases when either none or both of them are `null`), then it can stay as it is.
And I suspect in most cases it's true (unless permission is not granted, but this is caught in a separate exception).